diff --git a/cpp/model/ReceiptPrinter.h b/cpp/model/ReceiptPrinter.h
index 16c61e4..8f8db31 100644
--- a/cpp/model/ReceiptPrinter.h
+++ b/cpp/model/ReceiptPrinter.h
@@ -50,7 +50,7 @@ public:
 
     std::string presentDiscount(const Discount &discount) const
     {
-        std::string name = discount.getDescription() + "(" + discount.getProduct().getName() + ")";
+        std::string name = discount.getDescription() + " (" + discount.getProduct().getName() + ")";
         std::string pricePresentation = getFormattedNumberAsString(discount.getDiscountAmount(), 2);
         return formatLineWithWhitespace(name, pricePresentation);
     }
diff --git a/csharp/SupermarketReceipt/ReceiptPrinter.cs b/csharp/SupermarketReceipt/ReceiptPrinter.cs
index 954f9d7..451d481 100644
--- a/csharp/SupermarketReceipt/ReceiptPrinter.cs
+++ b/csharp/SupermarketReceipt/ReceiptPrinter.cs
@@ -51,7 +51,7 @@ namespace SupermarketReceipt
 
         private string PrintDiscount(Discount discount)
         {
-            string name = discount.Description + "(" + discount.Product.Name + ")";
+            string name = discount.Description + " (" + discount.Product.Name + ")";
             string value = PrintPrice(discount.DiscountAmount);
 
             return FormatLineWithWhitespace(name, value);
diff --git a/java/src/main/java/dojo/supermarket/ReceiptPrinter.java b/java/src/main/java/dojo/supermarket/ReceiptPrinter.java
index e6f1bbd..ac22b0f 100644
--- a/java/src/main/java/dojo/supermarket/ReceiptPrinter.java
+++ b/java/src/main/java/dojo/supermarket/ReceiptPrinter.java
@@ -45,7 +45,7 @@ public class ReceiptPrinter {
     }
 
     private String presentDiscount(Discount discount) {
-        String name = discount.getDescription() + "(" + discount.getProduct().getName() + ")";
+        String name = discount.getDescription() + " (" + discount.getProduct().getName() + ")";
         String value = presentPrice(discount.getDiscountAmount());
 
         return formatLineWithWhitespace(name, value);
diff --git a/python/receipt_printer.py b/python/receipt_printer.py
index 7550fe9..102916d 100644
--- a/python/receipt_printer.py
+++ b/python/receipt_printer.py
@@ -46,7 +46,7 @@ class ReceiptPrinter:
             return '%.3f' % item.quantity
 
     def print_discount(self, discount):
-        name = f"{discount.description} ({discount.product.name})"
+        name = f"{discount.description}  ({discount.product.name})"
         value = self.print_price(discount.discount_amount)
         return self.format_line_with_whitespace(name, value)
 
