diff --git a/cpp/model/ShoppingCart.cpp b/cpp/model/ShoppingCart.cpp
index 3c3a523..0421322 100644
--- a/cpp/model/ShoppingCart.cpp
+++ b/cpp/model/ShoppingCart.cpp
@@ -55,7 +55,7 @@ void ShoppingCart::handleOffers(Receipt& receipt, std::map<Product, Offer> offer
                 discount = new Discount(std::to_string(offer.getArgument()) + "% off", -quantity * unitPrice * offer.getArgument() / 100.0, product);
             }
             if (offer.getOfferType() == SpecialOfferType::FiveForAmount && quantityAsInt >= 5) {
-                double discountTotal = unitPrice * quantity - (offer.getArgument() * numberOfXs + quantityAsInt % 5 * unitPrice);
+                double discountTotal = unitPrice * quantity - (offer.getArgument() * numberOfXs + quantityAsInt % 6 * unitPrice);
                 discount = new Discount(std::to_string(x) + " for " + std::to_string(offer.getArgument()), -discountTotal, product);
             }
             if (discount != nullptr)
diff --git a/csharp/SupermarketReceipt/ShoppingCart.cs b/csharp/SupermarketReceipt/ShoppingCart.cs
index 28abbe7..90d472d 100644
--- a/csharp/SupermarketReceipt/ShoppingCart.cs
+++ b/csharp/SupermarketReceipt/ShoppingCart.cs
@@ -71,7 +71,7 @@ namespace SupermarketReceipt
                     if (offer.OfferType == SpecialOfferType.TenPercentDiscount) discount = new Discount(p, offer.Argument + "% off", -quantity * unitPrice * offer.Argument / 100.0);
                     if (offer.OfferType == SpecialOfferType.FiveForAmount && quantityAsInt >= 5)
                     {
-                        var discountTotal = unitPrice * quantity - (offer.Argument * numberOfXs + quantityAsInt % 5 * unitPrice);
+                        var discountTotal = unitPrice * quantity - (offer.Argument * numberOfXs + quantityAsInt % 6 * unitPrice);
                         discount = new Discount(p, x + " for " + offer.Argument, -discountTotal);
                     }
 
diff --git a/java/src/main/java/dojo/supermarket/model/ShoppingCart.java b/java/src/main/java/dojo/supermarket/model/ShoppingCart.java
index c03834d..f534c66 100644
--- a/java/src/main/java/dojo/supermarket/model/ShoppingCart.java
+++ b/java/src/main/java/dojo/supermarket/model/ShoppingCart.java
@@ -68,7 +68,7 @@ public class ShoppingCart {
                     discount = new Discount(p, offer.argument + "% off", -quantity * unitPrice * offer.argument / 100.0);
                 }
                 if (offer.offerType == SpecialOfferType.FiveForAmount && quantityAsInt >= 5) {
-                    double discountTotal = unitPrice * quantity - (offer.argument * numberOfXs + quantityAsInt % 5 * unitPrice);
+                    double discountTotal = unitPrice * quantity - (offer.argument * numberOfXs + quantityAsInt % 6 * unitPrice);
                     discount = new Discount(p, x + " for " + offer.argument, -discountTotal);
                 }
                 if (discount != null)
diff --git a/python/shopping_cart.py b/python/shopping_cart.py
index 4e36cc5..bfedad5 100644
--- a/python/shopping_cart.py
+++ b/python/shopping_cart.py
@@ -61,7 +61,7 @@ class ShoppingCart:
 
                 if offer.offer_type == SpecialOfferType.FIVE_FOR_AMOUNT and quantity_as_int >= 5:
                     discount_total = unit_price * quantity - (
-                                offer.argument * number_of_x + quantity_as_int % 5 * unit_price)
+                                offer.argument * number_of_x + quantity_as_int % 6 * unit_price)
                     discount = Discount(p, str(x) + " for " + str(offer.argument), -discount_total)
 
                 if discount:
